develop/tutorials/articles/192-item-selector/02-creating-custom-entites-for-the-item-selector.markdown

```diff
@@ -1,151 +1,131 @@
+                        _resourcePrimKey = resourcePrimKey;
+                }
+        
+                public long getResourcePrimKey() {
+                        return _resourcePrimKey;
+                }
+        
+                public void setResourcePrimKey(long resourcePrimKey) {
+                        _resourcePrimKey = resourcePrimKey;
+                }
+        
+                private long _resourcePrimKey;
+        
+        }
+
+    +$$$
+
+    **Note:** Criterion fields should be serializable and should expose a 
+    public empty constructor(as shown above).
+    
+    $$$
+
+2.  Create an OSGi component class that implements 
+    `BaseItemSelectorCriterionHandler`. Each criterion requires a criterion 
+    handler, which is responsible for obtaining the proper selection view.
+
+    The example below creates a criterion handler for the 
+    `TaskItemSelectorCriterion`:
+
+        @Component(service = ItemSelectorCriterionHandler.class)
+        public class TaskItemSelectorCriterionHandler extends 
+        BaseItemSelectorCriterionHandler<TaskItemSelectorCriterion> {
+
+            public Class <TaskItemSelectorCriterion> 
+            getItemSelectorCriterionClass() {
+                return TasksItemSelectorCriterionHandler.class;
+            }
+            
+            @Activate
+            @Override
+            protected void activate(BundleContext bundleContext) {
+                    super.activate(bundleContext);
+            }
+    
+        }
+        
+    The @Activate and @Override tokens are required to activate this OSGi 
+    component.
+    
+Depending on the needs of your app, you may not need to create a return type. If
+your entity returns information that is already defined by an existing return
+type, you can use that return type.
+
+You can view the default available criteria in the 
+[Item Selector Criterion and Return Types](/develop/reference/-/knowledge_base/7-0/item-selector-criterion-and-return-types) 
+reference.
+
+If, however, your entity returns information that is not covered by an existing 
+return type, you'll need to create a new return type next.
+
+## Creating Item Selector Return Types [](id=creating-item-selector-return-types)
+
+To create a return type, you must create a class that implements 
+`ItemSelectorReturnType`.
+
+The example below creates a new return type called `TaskItemSelectorReturnType`:
+
+    public class TasksItemSelectorReturnType implements ItemSelectorReturnType{
+    
+    }
+    
+Note that you can create a new return type for each piece of information that 
+your entity should return, but you can also create one return type to identify
+all the information if you wish. Like the `*ItemSelectorCriterion` class, **the 
+`*ItemSelectorReturnType` class is simply used as an identifier by the Item 
+Selector, and does not actually return any information itself.**
+
+So far, you've created an API that you can use to create a selection view for 
```

Comment:
I think that it's important to highlight that although the ItemSelectorReturnType is an identifier that doesn't return the information itself, it's an API, a contract between the client of the item selector and the item selector views.
That means that whenever that return type is used (even if it's in a completely different use case, with different item selector criteria and different item selector view), the view needs to ensure that it will return the same information in the same format specified in the return type. It's recommended to specify the information that the return type will return as well as the format as javadoc.